[Bug] Build fails with modern edk2: Tiano.h not found due to Base.h pragma once migration

## Summary

The UEFI SCT build fails with modern edk2 (latest main branch) because `SctPkg/Include/Efi.h` relies on the `__BASE_H__` macro to detect the new (EDK2) vs. old (EDK1/Tiano) build system. Modern edk2 `Base.h` switched from `#ifndef __BASE_H__` / `#define __BASE_H__` include guards to `#pragma once`, so `__BASE_H__` is never defined.

This causes `Efi.h` to fall into the legacy code path and attempt `#include "Tiano.h"`, which no longer exists (it was part of the long-deprecated EdkCompatibilityPkg).

## Error

```
In file included from SctPkg/Library/EfiTestLib/EfiTestLib.c:26:
SctPkg/Include/Efi.h:34:12: fatal error: Tiano.h: No such file or directory
   34 |   #include "Tiano.h"
      |            ^~~~~~~~~
```

This error blocks compilation of **every** SCT module since `Efi.h` is a foundational include.

## Root Cause

`SctPkg/Include/Efi.h` line 33-34:

```c
#ifndef __BASE_H__
  #include "Tiano.h"
```

Modern edk2 `MdePkg/Include/Base.h` uses `#pragma once` and does **not** define `__BASE_H__`. Therefore the `#ifndef` evaluates to true and the legacy (non-existent) header is included.

## Proposed Fix

Update the guard in `Efi.h` to also check for `MDE_CPU_*` macros, which are always defined by `ProcessorBind.h` (included by `Base.h`). This covers all supported architectures (IA32, X64, ARM, AARCH64, RISC-V, LoongArch64, EBC):

```c
#if !defined(__BASE_H__) && !defined(MDE_CPU_IA32) && !defined(MDE_CPU_X64) && \\
    !defined(MDE_CPU_EBC) && !defined(MDE_CPU_ARM) && !defined(MDE_CPU_AARCH64) && \\
    !defined(MDE_CPU_RISCV64) && !defined(MDE_CPU_LOONGARCH64)
```

This is backward compatible — the old `__BASE_H__` check still works for any legacy edk2 that defines it.

## Impact

- **Severity:** Build blocker for all SCT modules when using modern edk2
- **Affected files:** `SctPkg/Include/Efi.h` (1 file)
- **Risk:** Minimal — only adds additional OR conditions to an existing preprocessor guard

--- a/uefi-sct/SctPkg/Include/Efi.h
+++ b/uefi-sct/SctPkg/Include/Efi.h
@@ -2,6 +2,7 @@
 
   Copyright 2006 - 2015 Unified EFI, Inc.<BR>
   Copyright (c) 2010 - 2015, Intel Corporation. All rights reserved.<BR>
+  Copyright (c) 2026 Advanced Micro Devices, Inc. All rights reserved.<BR>
 
   This program and the accompanying materials
   are licensed and made available under the terms and conditions of the BSD License
@@ -29,8 +30,14 @@ Abstract:
 #ifndef _EFI_H_
 #define _EFI_H_
 
-// If we use the new build system
-#ifndef __BASE_H__
+// If we use the new build system.
+// Note: older edk2 Base.h used #define __BASE_H__ as its include guard;
+// modern edk2 uses #pragma once, so __BASE_H__ is no longer defined.
+// Check for MDE_CPU_* from ProcessorBind.h (always included by Base.h)
+// as a reliable indicator of the modern build system.
+#if !defined(__BASE_H__) && !defined(MDE_CPU_IA32) && !defined(MDE_CPU_X64) && \
+    !defined(MDE_CPU_EBC) && !defined(MDE_CPU_ARM) && !defined(MDE_CPU_AARCH64) && \
+    !defined(MDE_CPU_RISCV64) && !defined(MDE_CPU_LOONGARCH64)
   #include "Tiano.h"
   #include "PeiHob.h"
 

